### PR TITLE
Review of README.coding-style

### LINKED ISSUE #4960: Review README.coding-style
We've drifted from a max. recommended line width of 80 to 120 but the document doesn't reflect it. I saw a few typos in there as well.

BTW I thought we had a recommendation against Yoda speak like `if (1 == foo)` but apparently it didn't make it into that document. I guess I can live with it but my brain hiccups every time I need to read things in reverse order.

### PR DESCRIPTION
**Pull request by magnumripper**
Relaxed max. line width and placement of variable declarations.
Updated link to base document.
Also fix some typos and formatting.

Closes #4960

### CODE REVIEW

File: doc/README.coding-style [hunk 1]
```diff
@@ -147,20 +145,23 @@ This example is *bad*:
...
+3.2.4 When type casting, a space before '*' and/or after the right paren is
+      allowed but not imperative.  E.g:
+
+	byte_len = strlen((char *) utf8_string);
 
-E.g:
+        - or -
 
-	hostSalt = (cl_uint *) mem_alloc(SALT_BUFFER_SIZE);
+	byte_len = strlen((char*)utf8_string);
```
Comment (solardiz) [inline doc/README.coding-style]: For new code, I'd rather have us standardize on `(char *)utf8_string` - that is, space before the `*)`, no space after. This is consistent with e.g. `char *string;`.

**Review by solardiz — APPROVED** [review]
The changes look OK to me. I'm not so sure of the whole thing, but I have no time for it now. Thanks!

File: doc/README.coding-style (hunk 1 above)
Comment (magnumripper) [reply doc/README.coding-style]: I'm fine with that. My habit of dropping the other space is a remnant from the 80-column days

**Comment by magnumripper** [comment]
> I'm not so sure of the whole thing

Me neither. I'll merge it for now so it's at least not outdated but we might consider dropping it and try making [CONTRIBUTING.md](https://github.com/openwall/john/blob/bleeding-jumbo/CONTRIBUTING.md) self-contained (and much shorter than this).

**Pull request merged**